lightning/src/util/ser.rs

```diff
@@ -381,6 +383,39 @@ impl Readable for BigSize {
 	}
 }
 
+/// The lightning protocol uses u16s for lengths in most cases. As our serialization framework
+/// primarily targets that, we must as well. However, because we may serialize objects that have
+/// more than 65K entries, we need to be able to store larger values. Thus, we define a variable
+/// length integer here which is backwards-compatible but treats 0xffff as "read eight more bytes".
```

Comment:
s/which is backwards-compatible/that is backwards-compatible for < 0xffff

Reply:
Note the last part that says it's backwards compatible only for < 0xffff.

Reply:
I think if somebody is at exactly 0xffff, they're about to run into problems anyway.